### ISSUE TITLE
Test key simulator doesn't work with uppercase characters

### ISSUE THREAD

**Issue by matthew-carroll** [flutter/flutter#96021]
I'm using `WidgetTester`'s `sendKeyDownEvent` and `sendKeyEvent` to simulate hardware interaction with a custom text field implementation. For example, typing "Hello, World!" into the text field using individual hardware keyboard keys.

It looks like the keyboard simulator for tests does not support uppercase characters:
https://github.com/flutter/flutter/blob/5a862ebecdb14ad389cdce9cd7f79470ece27a7f/packages/flutter_test/lib/src/event_simulation.dart#L31

What I would expect is that if the shift key is currently pressed in the simulator, then the resolved character would be uppercase. This is also true for the various non-alpha-numeric characters on the keyboard as well.

Flutter version:
```
Flutter 2.8.1 • channel stable • git@github.com:flutter/flutter.git
Framework • revision 77d935af4d (2 weeks ago) • 2021-12-16 08:37:33 -0800
Engine • revision 890a5fca2e
Tools • Dart 2.15.1
```

**Comment by matthew-carroll** [comment]
CC @justinmc @gspencergoog

**Comment by darshankawar** [comment]
The key simulation according to current code implementation converts into `lowercase` as below:

https://github.com/flutter/flutter/blob/5a862ebecdb14ad389cdce9cd7f79470ece27a7f/packages/flutter_test/lib/src/event_simulation.dart#L34

And hence the current behavior.

**Comment by matthew-carroll** [comment]
Hi @darshankawar, I'm not sure if you're providing that future reference, or if you're suggesting that this is desired behavior. If the latter, I don't think that this should be the desired behavior. Flutter developers are limited to the testing tools provided in `WidgetTester` - those test tools should facilitate real world keyboard input, which includes uppercase and special characters.

**Comment by darshankawar** [comment]
@matthew-carroll Hi,
Thanks for the feedback. I just referred it as current limitation.

**Comment by matthew-carroll** [comment]
@darshankawar do you know which sub-team, or which team member this is likely to get assigned to? I'm concerned that there's been no feedback from any developer in a month.

**Comment by gspencergoog** [comment]
Yes, this is a limitation of the current key simulator code (see [my TODO](https://github.com/flutter/flutter/blob/5a862ebecdb14ad389cdce9cd7f79470ece27a7f/packages/flutter_test/lib/src/event_simulation.dart#L26) from a while ago). The intention is that this simulation code will be deprecated once we have a solution based on the new key event subsystem (e.g. the one using `KeyEvent` instead of `RawKeyEvent`).

cc @dkwingsmt who would be the one most likely to know the status of the new event simulation code.
